Add Topic model and Resources context

pedro-gaspar opened the pull request:
Laying the ground for starting collecting information around a particular topic.

All of the resources data management will live in the `Resources` context.

trenpixster commented on lib/curious_maria/resources/resources.ex:
```diff
@@ -0,0 +1,104 @@
+defmodule CuriousMaria.Resources do
```
I'm still not entirely sold on having this kind of Context shizzle when not developing DDD. It seems like a big buy in from a very very early start where domain is not clear. Anyhoo, who cares, guess it's just namespacing 😄

pedro-gaspar replied:
I read the recent phoenix book and they were using this context approach. So my mindset is, if this is how the cool kids are working these days, well let me try it then.

You are right this is just one more abstraction.

Like Repositories in ASP.NET MVC, all of a sudden everyone started using for tests. 
And Services in Ruby on Rails.

The first time I saw MVC in Ruby on Rails I was like, WTF so many files. In .net we do a click handler and then do the stuff, way simpler. But it baked and nowadays it's the defacto way so whatever, lets see :smile:

pedro-gaspar commented:
You are absolutely right. It would be better to generate and then do a separate commit with the changes to highlight what's different. 

Short story, I didn't do a lot of changes, just moved function to create a fixture to a `FixtureHelpers` module, because it will probably be needed when doing other tests.

https://github.com/pedro-gaspar/curious_maria/blob/042117338e589ec7bb233583d171815d0515e07f/test/support/fixture_helpers.ex#L6

And removed the required validation of the description field.

https://github.com/pedro-gaspar/curious_maria/blob/042117338e589ec7bb233583d171815d0515e07f/lib/curious_maria/resources/topic.ex#L21

pedro-gaspar commented:
@trenpixster Refactored this pull request with the same points you mentioned in #3

filipecabaco commented on test/support/fixture_helpers.ex:
```diff
@@ -0,0 +1,14 @@
+  alias CuriousMaria.{Resources}
+
+  @topic_attrs %{title: "some title", description: "some description"}
+
```
```suggestion
  defp rand_string(), do: Base.url_encode64(:crypto.strong_rand_bytes(20))
  defp generate_generate_fixture(), do: %{title: rand_string(), description: rand_string()}
  def topic_fixture(attrs \\ %{}) do
    {:ok, topic} =
      attrs
      |> Enum.into(generate_generate_fixture())
      |> Resources.create_topic()
     topic
  end
```

filipecabaco replied:
yes 👍 when possible, IMO, using random values will assure you're not testing your mocks. I found bugs due to this approach... it's like a really lite and wrong version of property base testing

filipecabaco replied:
😅
